tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleShareConsumerOptions.java

```diff
@@ -0,0 +1,268 @@
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.MessageFormatter;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.server.util.CommandDefaultOptions;
+import org.apache.kafka.server.util.CommandLineUtils;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public final class ConsoleShareConsumerOptions extends CommandDefaultOptions {
+    private final OptionSpec<String> messageFormatterOpt;
+    private final OptionSpec<String> messageFormatterConfigOpt;
+    private final OptionSpec<String> messageFormatterArgOpt;
+    private final OptionSpec<String> keyDeserializerOpt;
+    private final OptionSpec<String> valueDeserializerOpt;
+    private final OptionSpec<Integer> maxMessagesOpt;
+    private final OptionSpec<?> rejectMessageOnErrorOpt;
+    private final OptionSpec<?> rejectOpt;
+    private final OptionSpec<?> releaseOpt;
+    private final OptionSpec<String> topicOpt;
+    private final OptionSpec<Integer> timeoutMsOpt;
+    private final OptionSpec<String> bootstrapServerOpt;
+    private final OptionSpec<String> groupIdOpt;
+    private final Properties consumerProps;
+    private final MessageFormatter formatter;
+
+    public ConsoleShareConsumerOptions(String[] args) throws IOException {
+        super(args);
+        topicOpt = parser.accepts("topic", "The topic to consume on.")
+                .withRequiredArg()
+                .describedAs("topic")
+                .ofType(String.class);
+        OptionSpec<String> consumerPropertyOpt = parser.accepts("consumer-property", "A mechanism to pass user-defined properties in the form key=value to the consumer.")
+                .withRequiredArg()
+                .describedAs("consumer_prop")
+                .ofType(String.class);
+        OptionSpec<String> consumerConfigOpt = parser.accepts("consumer.config", "Consumer config properties file. Note that " + consumerPropertyOpt + " takes precedence over this config.")
+                .withRequiredArg()
+                .describedAs("config file")
+                .ofType(String.class);
+        messageFormatterOpt = parser.accepts("formatter", "The name of a class to use for formatting Kafka messages for display.")
+                .withRequiredArg()
+                .describedAs("class")
+                .ofType(String.class)
+                .defaultsTo(DefaultMessageFormatter.class.getName());
+        messageFormatterArgOpt = parser.accepts("property",
+                        "The properties to initialize the message formatter. Default properties include: \n" +
+                                " print.timestamp=true|false\n" +
+                                " print.key=true|false\n" +
+                                " print.offset=true|false\n" +
+                                " print.delivery=true|false\n" +
```

Comment:
Do we have a new formatter which accepts `print.delivery`?

Reply:
But my question is that should it be included in `DefaultMessageFormatter` as there I see the usage.

Reply:
Probably best to remove it for now. I'll add it back in when I do `ConsumerRecord.deliveryCount()`.

Reply:
Yeah, agree. Worth to remove if we do not have any use of it yet.

Reply:
Okay yeah, I have removed the line now.